decidim-core/lib/decidim/resourceable.rb

```diff
@@ -82,6 +82,28 @@ def link_resources(resources, link_name, data = {})
       end
 
       delegate :resource_manifest, to: :class
+
+      # Checks throughout all its parent hierarchy if this Resource is visible.
+      # i.e. checks
+      # - the visibility of its parent Component
+      # - the visibility of its participatory space.
+      # - the visibility of the resource itself.
+      def visible?
```

Comment:
I think `visible` might be a bit ambiguous. Visible when? For whom?

What about `public?` as a name instead? Sorry about being too nitpicky with this, but since it will grow into an internal API, finding appropiate names is usually a good investment.

Reply:
@mrcasals @oriolgual thoughts?

Reply:
`public` is the contrary of `private`, a concept that already exists. In fact, `(private == false) == public`, non-private is the same as public. But there are other attributes that may affect the "visibility" to the resource, like published. For me using public will create confusion.

On the other side, responding to your questions, "when?" and "for whom?" (you can ask the same for public).
- When? at the time of asking, or when you invoke the method if you prefer.
- For whom? `visible?` does not take into account the specific permissions for a given user, for this, I already proposed a `visible_for?(user)` method. `visible?` will check the global visibility (all influencing attributes except user specific) of a resource.

Alternatives to "visible" may be: accessible, disposable, able_to_use, able_to_access, can_manage, can_use, can_see. Names are important but conventions are important the same, I think that it will be practical to use "visible" as a convention, I can document it better.

Reply:
Hmmm, okay, you're right 👍. Colliding with `public` is not a good idea. Let's merge this!